Sources/StreamChat/Repositories/ConnectionRepository.swift

```diff
@@ -95,14 +99,6 @@ class ConnectionRepository {
             return
         }
 
-        if connectionId == nil {
-            if source == .userInitiated {
-                log.warning("The client is already disconnected. Skipping the `disconnect` call.")
-            }
-            completion()
-            return
-        }
```

Comment:
This does not make sense, it is unnecessary optimization that just complicates things. For example, when we timeout, we don't have the connectionId, so the `webSocketClient? .disconnect` won't be called. Which means no status changes will be reported.